content/developer/reference/frontend/patching_code.rst

```diff
@@ -178,19 +156,48 @@ can easily be patched as well (see the section on :ref:`best practices<frontend/
 Removing a patch
 ================
 
-The `patch` function has a counterpart, `unpatch`, also located in `@web/core/utils/patch`.
+The `patch` function returns its counterpart. This is mostly useful for
+testing purposes, when we patch something at the beginning of a test, and
+unpatch it at the end.
+
+.. code-block:: javascript
+
+    const unpatch = patch(object, { ... });
+    // test stuff here
+    unpatch();
+
+Applying a same patch to multiple objects
+=========================================
 
-.. js:function:: unpatch(obj, patchName)
+It could happen that one wants to apply a same patch to multiple object but because
+of the way `super` keyword works, the `extension` has to be unique by patch and
+cannot be copied/cloned (`check the documentation of the keyword <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/super#description>`_).
+A function returning the object used to patch can be used to make it unique.
 
-    :param Object obj: object that should be unpatched
-    :param string patchName: string describing the patch that should be removed
+.. code-block:: javascript
+
+    const obj1 = {};
+    const obj2 = {};
+
+    function createExtensionObj() {
+      return { ... };
+    }
+
+    patch(obj1, createExtensionObj());
+    patch(obj2, createExtensionObj());
+
+.. warning::
+
+  If an `extension` is based on another then the two extensions should
+  be applied separately. Do not copy/clone an extension.
+
+  .. code-block:: javascript
 
-    Removes an existing patch from an object `obj`. This is mostly useful for
-    testing purposes, when we patch something at the beginning of a test, and
-    unpatch it at the end.
+      const object = {};
 
-    .. code-block:: javascript
+      const ext1 = { a: 1 };
+      const invalid_ext2 = { ...ext1, b: 2 };
```

Comment:
add a comment (eg `// this will not work: super will not refer to the correct object in methods coming from ext1`